### PR TITLE
Fix Compatibility Issues

### PR DESCRIPTION
**Pull request by cdmihai**
This refers to the compatibility report in issue #299.

Makes FixCommandLineSwitch internal and adjusts a test in order to avoid namespaces conflicts (that would have been caused by an internalVisibleTo)

### CODE REVIEW

**Comment by rainersigwald** [comment]
LGTM.  @ValMenn might have opinions about the test change.

**Comment by dsplaisted** [comment]
Can you add the APIs from Update 1 to compatbaseline.txt (with a comment explaining why) so the compatibility report will be clean?

**Comment by AndyGerlicher** [comment]
LGTM

**Comment by ValMenn** [comment]
We discussed the change with Mihai. LGTM

File: CompatBaseline.txt [hunk 1]
```diff
@@ -1,2 +1,8 @@
 # This API is marked obsolete and has been removed from the reference assembly
-MembersMustExist : Member 'Microsoft.Build.Utilities.ToolTask.EnvironmentOverride.get()' does not exist in the implementation but it does exist in the contract.
\ No newline at end of file
+MembersMustExist : Member 'Microsoft.Build.Utilities.ToolTask.EnvironmentOverride.get()' does not exist in the implementation but it does exist in the contract.
```
Comment (cdmihai) [inline CompatBaseline.txt]: @dsplaisted I updated the CompatBaseline file. Now there are no difference reported.
Does this look good to you?

File: CompatBaseline.txt [hunk 2]
```diff
@@ -1,2 +1,8 @@
 # This API is marked obsolete and has been removed from the reference assembly
-MembersMustExist : Member 'Microsoft.Build.Utilities.ToolTask.EnvironmentOverride.get()' does not exist in the implementation but it does exist in the contract.
```
Comment (cdmihai) [inline CompatBaseline.txt]: @dsplaisted I updated the CompatBaseline file. Now there are no difference reported.
Does this look good to you?

**Pull request merged**